phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -911,7 +911,8 @@ private void addCoprocessors(byte[] tableName, TableDescriptorBuilder builder, P
                     QueryServicesOptions.DEFAULT_INDEX_REGION_OBSERVER_ENABLED);
 
             if (tableType == PTableType.INDEX && !isTransactional) {
-                if (globalIndexerEnabled && !newDesc.hasCoprocessor(GlobalIndexChecker.class.getName())) {
+                if (globalIndexerEnabled && !newDesc.hasCoprocessor(GlobalIndexChecker.class.getName()) &&
+                        !isLocalIndexTable(builder.build().getColumnFamilyNames())) {
```

Comment:
nit: can use newDesc for the TableDescriptor rather than using builder.build() to reconstruct it.